tycho-versions-plugin/src/main/java/org/eclipse/tycho/versions/pom/PomFile.java

```diff
@@ -158,13 +167,35 @@ public void setParentVersion(String newVersion) {
      */
     public void setVersion(String version) {
         this.version = version;
+        this.resolvedVersion = null;
     }
 
     /**
      * Returns the (effective) version of the project.
      */
     public String getVersion() {
-        return version;
+        if (this.resolvedVersion == null) {
+            this.resolvedVersion = resolveCiFriendlyVersion();
+        }
+        return this.resolvedVersion;
+    }
+
+    /**
+     * Returns the list of property names that make up the <code>&lt;version&gt;</code> of the
+     * project.
+     * <p>
+     * If this list is non-empty, the project uses ci-friendly versions.
+     */
+    public List<String> getCiFriendlyVersionPropertyNames() {
```

Comment:
Maybe move to `PomVersionsUtil` and pass the version as a parameter?
```suggestion
    public static List<String> getCiFriendlyVersionPropertyNames(String version) {
```